viewer-prototype/src/node/trace-server-service.ts

```diff
@@ -4,11 +4,24 @@ import { TraceServerConfigService } from '../common/trace-server-config';
 @injectable()
 export class TraceServerServiceImpl implements TraceServerConfigService {
 
-    async startTraceServer(path: string | undefined, port: number | undefined): Promise<void> {
-        spawn(`${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
+    startTraceServer(path: string | undefined, port: number | undefined): Promise<void> {
+        const server = spawn(`${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
```

Comment:
Spawning the child process will start the server, if it's success full, the server process will never exit. So, I don't think we will be able to get a resolve beside the timeout. I actually notice that when running it on my laptop. I get a timeout in the calling method and the error dialog is shown, while there server is actually running. We need to find another way to determine that the server is actually started successfully.

Reply:
Yes, determining 'success' is difficult here. I think the best we can do is detect whether the process emits output of any kind, and resolve if it does. This means that it will reject if the spawn itself produces an error, but it would struggle with a server that started and ran completely silently.